packages/appium/docs/scripts/gen-cli-args-docs.js

```diff
@@ -0,0 +1,95 @@
+// for simplicity this file is not transpiled and is run directly via an npm script
+/* eslint-disable no-console */
+/* eslint-disable promise/prefer-await-to-callbacks */
+/* eslint-disable promise/prefer-await-to-then */
+
+const path = require('path');
+const {fs} = require('@appium/support');
+const {AppiumConfigJsonSchema} = require('@appium/schema');
+
+const DOC_PATH = path.resolve(__dirname, '..', 'en', 'cli', 'args.md');
+
+const START_TAG = `<!-- AUTOGEN-START -->`;
+const STOP_TAG = `<!-- AUTOGEN-STOP -->`;
+
+async function main() {
+  console.log(`Gathering current cli doc file contents...`);
+  const [preStart, postEnd] = await getDocParts();
+  console.log(`Generating new cli arg table...`);
+  const argTable = getCliArgTable();
+  console.log(`Writing new doc file...`);
+  await writeNewDoc(preStart, argTable, postEnd);
+  console.log(`...done!`);
+}
+
+async function getDocParts() {
+  if (!(await fs.exists(DOC_PATH))) {
+    throw new Error(`Cannot find args.md at ${DOC_PATH}`);
+  }
+
+  const curContents = await fs.readFile(DOC_PATH, 'utf-8');
+
+  // attempt to verify the start/stop tags
+  const [preStart, postStart] = curContents.split(START_TAG);
+  if (!postStart) {
+    throw new Error(`Could not find autogen start tag '${START_TAG}'. Make sure it's in place!`);
+  }
+  const [, postEnd] = postStart.split(STOP_TAG);
+  if (!postEnd) {
+    throw new Error(`Could not find autogen stop tag '${STOP_TAG}'. Make sure it's in place!`);
+  }
+
+  return [preStart, postEnd];
+}
+
+function getCliArgTable() {
+  const pad = '&nbsp;'.repeat(30);
+  const header = `|Argument${pad}|Description|Type|Default|Aliases|\n` + `|--|--|--|--|--|`;
+  const serverProps = AppiumConfigJsonSchema.properties.server.properties;
+  const args = Object.keys(serverProps).map((arg) => {
+    let {appiumCliAliases = [], description, type, items, maximum, minimum} = serverProps[arg];
+    let def = serverProps[arg].default;
+    let enums = serverProps[arg].enum;
+    if (type === 'array' || type === 'object' || (type === 'string' && def === '')) {
+      def = JSON.stringify(def);
+    }
+    if (type === 'array' && items.type) {
+      type = `array<${items.type}>`;
+    }
+    if (type === 'integer' && typeof minimum !== 'undefined' && typeof maximum !== 'undefined') {
+      description = `${description} (Value must be between \`${minimum}\` and \`${maximum}\`)`;
```

Comment:
can it be that only minimum or maximum is defined?

Reply:
dunno, I just made sure we could handle everything I found in the current schema with a degree of niceness. i didn't dig into all the schema possibilities. basically if there are other things we could do, it shouldn't break this code. but it could be an opportunity for more niceness in a future update.

Reply:
@mykola-mokhnach how does that sound to you?

Reply:
seems reasonable